### PR TITLE
Netplan integration

### PR DESCRIPTION
**Pull request by eroussy**
Integrate netplan to configure
- admin network
- ptp network
- cluster network IP adresses and OVS bridge

The additionnal vxlan bridge cannot be configured with OVS.

The only API change of this PR is the variable `remove_all_networkd_config` renamed to `remove_all_network_config`

### CODE REVIEW

File: inventories/netplan_cluster_example.yaml.j2 [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# This netplan files defines the path of the cluster network which is managed
+# by systemd. Especially, it attributes its cluster IP addr to each node.
+network:
+  version: 2
```
Comment (dupremathieu) [inline inventories/netplan_cluster_example.yaml.j2:4]: Defined the full Open vSwitch team0 bridge here.
Refer to: https://github.com/canonical/netplan/blob/main/examples/openvswitch.yaml

**Review by dupremathieu — CHANGES_REQUESTED** [review]

**Review by dupremathieu — CHANGES_REQUESTED** [review]
I will add a Netplan example file for an extended OVS bridges.

File: inventories/netplan_cluster_example.yaml.j2 [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Copyright (C) 2024, SFL (https://savoirfairelinux.com)
+# SPDX-License-Identifier: Apache-2.0
+# This netplan files defines the path of the cluster network which is managed
+# by systemd. Especially, it attributes its cluster IP addr to each node.
+network:
+  version: 2
+  renderer: networkd
+  ethernets:
+    "{{ team0_0|default('0') }}":
+      mtu: 1800
```
Comment (dupremathieu) [inline inventories/netplan_cluster_example.yaml.j2:10]: Why set the MTU to 1800?

**Review by dupremathieu — CHANGES_REQUESTED** [review]

**Comment by eroussy** [comment]
> I will add a Netplan example file for an extended OVS bridges.

This is not possible currently with netplan.

File: inventories/netplan_cluster_example.yaml.j2 (hunk 2 above)
Comment (insatomcat) [reply inventories/netplan_cluster_example.yaml.j2]: team0 need to have an increased MTU because of vxlan running over it.

Comment (eroussy) [reply inventories/netplan_cluster_example.yaml.j2:10]: Done

Comment (dupremathieu) [reply inventories/netplan_cluster_example.yaml.j2:10]: There is no vxlan with team0.
Recommended MTU for Ceph is 9000: https://docs.redhat.com/en/documentation/red_hat_ceph_storage/4/html/configuration_guide/ceph-network-configuration#verifying-and-configuring-the-mtu-value_conf

I suggest to use this value.

**Review by dupremathieu — CHANGES_REQUESTED** [review]

**Review by dupremathieu — APPROVED** [review]

**Pull request merged**